File: 3255/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(resultsArray([]int{3,2,3,2,3,2}, 2))
}

func resultsArray(nums []int, k int) []int {
	n := len(nums)
	cnt := 0 
	ans := make([]int, n-k+1)
	for i := 0; i < n; i++ {
		if i == 0 || nums[i] - nums[i-1] != 1 {
			cnt = 1
		} else {
			cnt++
		}
		if cnt >= k {
			ans[i-k+1] = nums[i]
		} else if i - k + 1 >= 0 {
			ans[i-k+1] = -1
		}
	}
	return ans
}

File: 3255/main.rs
struct Solution {}

impl Solution {
    pub fn results_array(nums: Vec<i32>, k: i32) -> Vec<i32> {
        let n = nums.len();
        let mut cnt = 0;
        let k = k as usize;
        let mut ans = vec![-1; n-k+1];
        for i in 0..n {
            if i == 0 || nums[i] - nums[i-1] != 1 {
                cnt = 1
            } else {
                cnt += 1
            }
            if cnt >= k {
                ans[i-k+1] = nums[i]
            }
        }
        ans
    }
}

fn main() {
    println!("{:#?}", Solution::results_array(vec![3,2,3,2,3,2], 2));
}